File: echo-guardian-production-foundation-v1.0/native/android/echo-guardian-core/src/main/java/org/echoguardian/accessibility/AccessibilityFlowModel.kt
package org.echoguardian.accessibility

data class AccessibilityScreen(
    val id: String,
    val title: String,
    val plainLanguage: String,
    val primaryAction: String,
    val voicePrompt: String,
)

object AccessibilityFlowFactory {
    fun defaultScreens(): List<AccessibilityScreen> = listOf(
        AccessibilityScreen("status", "Safety Status", "This screen tells you if Echo Guardian is watching for safety changes.", "Ask what are you doing", "What are you doing?"),
        AccessibilityScreen("confirmation", "Are you okay?", "Echo Guardian is checking if you are okay. You can answer by voice, text, or touch.", "I am okay", "Are you okay?"),
        AccessibilityScreen("alert_preview", "Alert Preview", "This shows exactly what your safety contact would receive.", "Review alert", "Read the alert preview")
    )
}

File: echo-guardian-production-foundation-v1.0/native/ios/EchoGuardianKit/Sources/EchoGuardianKit/Accessibility/AccessibilityFlowModel.swift
import Foundation

public struct AccessibilityScreen: Codable, Equatable {
    public let id: String
    public let title: String
    public let plainLanguage: String
    public let primaryAction: String
    public let voicePrompt: String
}

public enum AccessibilityFlowFactory {
    public static func defaultScreens() -> [AccessibilityScreen] {
        [
            AccessibilityScreen(id: "status", title: "Safety Status", plainLanguage: "This screen tells you if Echo Guardian is watching for safety changes.", primaryAction: "Ask what are you doing", voicePrompt: "What are you doing?"),
            AccessibilityScreen(id: "confirmation", title: "Are you okay?", plainLanguage: "Echo Guardian is checking if you are okay. You can answer by voice, text, or touch.", primaryAction: "I am okay", voicePrompt: "Are you okay?"),
            AccessibilityScreen(id: "alert_preview", title: "Alert Preview", plainLanguage: "This shows exactly what your safety contact would receive.", primaryAction: "Review alert", voicePrompt: "Read the alert preview")
        ]
    }
}
